snapcraft/internal/meta/_desktop.py

```diff
@@ -35,7 +35,7 @@ def __init__(
         self._prime_dir = prime_dir
         self._path = os.path.join(prime_dir, filename)
         if not os.path.exists(self._path):
-            raise errors.InvalidDesktopFileError(
+            raise errors.MissingDesktopFileError(
                 filename, 'does not exist (defined in the app {!r})'.format(
                     name))
```

Comment:
Take a look at this exception. It now accepts a filename, and an app name. You're still passing it a filename and a message.

Reply:
kyrofa..does that mean that i should remove the message and the filename..that is line 39 and 40 completely?